crates/configuration/src/parachain.rs

```diff
@@ -833,6 +833,8 @@ impl<C: Context> ParachainConfigBuilder<WithId, C> {
             self.default_chain_context(),
             self.validation_context.clone(),
         ))
+        .validator(true)
+        .with_args(vec!["--collator".into()])
```

Comment:
If the `node` is set as validator we inject automatically the `--collator` flag.
```suggestion
```